configs/ts-config/tsconfig.base.json

```diff
@@ -0,0 +1,17 @@
+{
+  "compilerOptions": {
+    "allowSyntheticDefaultImports": true,
+    "allowUnreachableCode": false,
+    "declaration": false,
+    "esModuleInterop": true,
+    "experimentalDecorators": true,
+    "lib": ["dom", "dom.iterable", "ES2021"],
+    "module": "esnext",
+    "moduleResolution": "node",
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "resolveJsonModule": true,
+    "target": "ES2021"
+  },
+  "exclude": ["node_modules", "src", "dist"]
+}
```

Comment:
This PR adds two separate `tsconfig.base.json` files. I assume we want the one in `packages/calcite-design-tokens` for this PR right? And then the move to a shared config will happen in #7753

Reply:
oops. that config work wasn't supposed to end up in this PR.